### PR TITLE
tmaurer3/check_for_nan

### PR DESCRIPTION
**Pull request by tmaurer3**
@lucianpls please review. 

Still not 100% sure whether we should add this check for NaN values or not. Feels a bit like an overkill.

### CODE REVIEW

**Comment by lucianpls** [comment]
The changes solve the problem with NaNs, but infinities can still cause problems.
It also requires C++11, which is probably not a significant issue.

**Comment by lucianpls** [comment]
Ups, I didn't mean to close it!

File: src/LercTest/main.cpp [hunk 1]
```diff
@@ -58,6 +58,7 @@ int main(int argc, char* arcv[])
     {
       zImg[k] = sqrt((float)(i * i + j * j));    // smooth surface
       zImg[k] += rand() % 20;    // add some small amplitude noise
+      //zImg[k] = NAN;
```
Comment (rouault) [inline src/LercTest/main.cpp]: debug trace

File: src/LercLib/Lerc.cpp [hunk 2]
```diff
@@ -462,3 +474,47 @@ ErrCode Lerc::ConvertToDoubleTempl(const T* pDataIn, size_t nDataValues, double*
 
 // -------------------------------------------------------------------------- ;
 
+template<class T> static ErrCode Lerc::CheckForNaN(const T* arr, int nDim, int nCols, int nRows, const BitMask* pBitMask)
+{
+  if (!arr || nDim <= 0 || nCols <= 0 || nRows <= 0)
+    return ErrCode::WrongParam;
+
+  const bool isDbl = (typeid(*arr) == typeid(double));
```
Comment (rouault) [inline src/LercLib/Lerc.cpp]: I guess there would be a more template'ish way of making sure this method is only defined for float and double, and a no-op for other types. Which would save the isFlt ternary tests and cast below

**Comment by tmaurer3** [comment]
To finish this, added a define switch for it which is turned off by default.

**Pull request merged**